LPS-59533

mhan810 opened the pull request.

mhan810 commented:
@hhuijser can you take a look @ this pull and see why source formatter keeps stripping out the Language.properties?

I'm moving JSPs from the portal-settings to portal-ldap.  However, SF keeps removing the properties I added to portal-ldap/src/resources/content/Language.properties and adding them to portal-impl/src/content/Language,properties.

Once it does that it then complains that I'm missing language keys which are already in place.

Cheers

liferay-continuous-integration commented:
<h1>Some tests FAILED!</h1><p>Build Time: 38 minutes 22 seconds</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br />Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/a617ca42ad4b8e11162119e0e89bc9df1699b250">a617ca42ad4b8e11162119e0e89bc9df1699b250</a></p><h4>Job Summary:</h4><ul><li><strike><strong><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest(master)/380/">test-portal-acceptance-pullrequest(master)</a></strong></strike></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/1834/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat-hypersonic/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/1820/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat-mysql/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12415/">test-portal-acceptance-pullrequest-batch(master)/integration-db2</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12432/">test-portal-acceptance-pullrequest-batch(master)/integration-hypersonic</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12523/">test-portal-acceptance-pullrequest-batch(master)/integration-mysql</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12392/">test-portal-acceptance-pullrequest-batch(master)/integration-oracle</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12506/">test-portal-acceptance-pullrequest-batch(master)/integration-postgresql</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/5711/">test-portal-acceptance-pullrequest-batch(master)/integration-sybase</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12424/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-db2</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/3977/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-hypersonic</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/11611/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4211/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-oracle</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/3940/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-postgresql</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12514/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-sybase</a></li><li><strike><strong><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4187/">test-portal-acceptance-pullrequest-batch(master)/modules-unit</a></strong></strike></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/5594/">test-portal-acceptance-pullrequest-batch(master)/plugins</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4209/">test-portal-acceptance-pullrequest-batch(master)/portal-web</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4656/">test-portal-acceptance-pullrequest-batch(master)/ruby-sass-compiler</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/5484/">test-portal-acceptance-pullrequest-batch(master)/service-builder</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4678/">test-portal-acceptance-pullrequest-batch(master)/tck</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/1835/">test-portal-acceptance-pullrequest-batch(master)/unit</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-source(master)/376/">test-portal-acceptance-pullrequest-source(master)</a></li></ul><h5>For more details click <a href="https://test-1-16.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/380/jenkins-report.html">here</a>.</h5><hr /><h4>Failed Jobs:</h4><ol><li><h5><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest(master)/380/">test-portal-acceptance-pullrequest(master)</a></h5><h6>Job Results:</h6><p>21 Jobs Passed.<br />2 Jobs Failed.</p><pre>Completed with the status of FAILURE.</pre></li><li><h5 job-result="UNSTABLE"><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4187/">test-portal-acceptance-pullrequest-batch(master)/modules-unit</a></h5><h6>Job Results:</h6><p>1123 Tests Passed.<br />1 Test Failed.</p><ol><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch%28master%29/AXIS_VARIABLE=3,label_exp=!master/4187//testReport/com.liferay.source.formatter/SourceFormatterTest/testSourceFormatter">SourceFormatterTest.testSourceFormatter</a></li></ol></li></ol>

hhuijser commented:
Hey @mhan810 

For those keys, since you only added the keys to portal-ldap (and not removed (yet) from portal-settings), we have duplicate keys in those two modules, and SF consolidates those in portal-impl.
After you remove from portal-settings (if you are sure you don't need those there anymore), you should be fine.

I don't understand how you would get complaints about missing keys when they got moved to portal-impl. Where did you run SF from to get that?

mhan810 commented:
K, I’ll try to remove the ones from portal-settings and see what happens.



I ran the SF from portal-impl (liferay-portal/portal-impl).  So the order
of events was:
(1) SF ran against portal-ldap and stripped from Language.properties there.

(2) SF ran against portal-settings and stripped from Language.properties
there.

(3) SF ran against portal-impl and added the language keys there.



See log:



E:\projects\liferay\trunk\liferay-portal\portal-impl>ant format-source

Buildfile: E:\projects\liferay\trunk\liferay-portal\portal-impl\build.xml



format-source:

     [java] Loading
file:/E:/projects/liferay/trunk/liferay-portal/portal-impl/classes/system.properties

     [java]
.\modules\apps\portal-settings\portal-settings-web\src\main\resources\content\Language.properties

     [java]
.\modules\portal\portal-ldap\src\main\resources\content\Language.properties

     [java] portal-impl\src\content\Language.properties

     [java] missing language key: add-ldap-server
./modules/apps/portal-settings/portal-settings-web/src/main/resources/META-INF/resources/edit_ldap_server.jsp



*From:* Hugo Huijser [mailto:notifications@github.com]
*Sent:* Sunday, November 1, 2015 7:07 PM
*To:* hhuijser/liferay-portal <liferay-portal@noreply.github.com>
*Cc:* Michael Han <michael.han@liferay.com>
*Subject:* Re: [liferay-portal] LPS-59533 (#2505)



Hey @mhan810 <https://github.com/mhan810>

For those keys, since you only added the keys to portal-ldap (and not
removed (yet) from portal-settings), we have duplicate keys in those two
modules, and SF consolidates those in portal-impl.
After you remove from portal-settings (if you are sure you don't need those
there anymore), you should be fine.

I don't understand how you would get complaints about missing keys when
they got moved to portal-impl. Where did you run SF from to get that?

—
Reply to this email directly or view it on GitHub
<https://github.com/hhuijser/liferay-portal/pull/2505#issuecomment-152884949>
.
